pkg/firewall/firewall_test.go

```diff
@@ -320,16 +328,16 @@ func TestCachesAllActiveKeepMembers(t *testing.T) {
 
 	policy.Validate(key.NetworkKeyToECDSAKey(remotePeerPublicKey))
 
-	if !cache.Has(activeKeepMembers[0].String()) {
+	if !activeKeepMembersCache.Has(activeKeepMembers[0].String()) {
 		t.Errorf("should cache active keep members")
 	}
-	if !cache.Has(activeKeepMembers[1].String()) {
+	if !activeKeepMembersCache.Has(activeKeepMembers[1].String()) {
 		t.Errorf("should cache active keep members")
 	}
-	if cache.Has(closedKeepMembers[0].String()) {
+	if activeKeepMembersCache.Has(closedKeepMembers[0].String()) {
 		t.Errorf("should not cache non-active keep members")
 	}
-	if cache.Has(closedKeepMembers[1].String()) {
+	if activeKeepMembersCache.Has(closedKeepMembers[1].String()) {
 		t.Errorf("should not cache non-active keep members")
 	}
```

Comment:
Shouldn't we add assertions for `noActiveKeepMembersCache`?

Reply:
The `noActiveKeepMembersCache` works a bit differently. We can't put members of an inactive keep inside it because those members can be part of an other active keep. Because of that, just the checked member may be inserted into the `noActiveKeepMembersCache` so making similar assertons for this cache here is not possible.

Reply:
Right. This is what I propose to assert. We can assert neither `activeKeepMembers` nor `closedKeepMembers` are in that cache but also that the guy for which we are executing the check - `key.NetworkPubKeyToEthAddress(remotePeerPublicKey)` - is there:
```
	// We don't put members of an active keep inside noActiveKeepMembersCache.
	// We don't put members of an inactive keep inside noActiveKeepMembersCache
	// because those members can belong to another active keep. We put there
	// only those members for which we executed the check.
	if noActiveKeepMembersCache.Has(activeKeepMembers[0].String()) {
		t.Errorf("should not cache member until all keeps are checked")
	}
	if noActiveKeepMembersCache.Has(activeKeepMembers[1].String()) {
		t.Errorf("should not cache member until all keeps are checked")
	}
	if noActiveKeepMembersCache.Has(closedKeepMembers[0].String()) {
		t.Errorf("should not cache member until all keeps are checked")
	}
	if noActiveKeepMembersCache.Has(closedKeepMembers[1].String()) {
		t.Errorf("should not cache member until all keeps are checked")
	}
	if !noActiveKeepMembersCache.Has(key.NetworkPubKeyToEthAddress(remotePeerPublicKey)) {
		t.Errorf("should be in the no active keep members cache")
	}
```